File: apis/src/main/java/com/beat/apis/performance/application/dto/performanceDetail/PerformanceDetailStaffResponse.java
package com.beat.apis.performance.application.dto.performanceDetail;

import com.beat.global.support.jackson.CdnImageUrl;

public record PerformanceDetailStaffResponse(
	Long staffId,
	String staffName,
	String staffRole,
	@CdnImageUrl String staffPhoto
) {
	public static PerformanceDetailStaffResponse of(Long staffId, String staffName, String staffRole,
		String staffPhoto) {
		return new PerformanceDetailStaffResponse(staffId, staffName, staffRole, staffPhoto);
	}
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/PerformanceDetailStaffResponse.kt
package com.beat.apis.performance.api.response

import com.beat.application.frontoffice.performance.StaffResult
import com.beat.global.support.jackson.CdnImageUrl

@ConsistentCopyVisibility
data class PerformanceDetailStaffResponse private constructor(
    val staffId: Long?,
    val staffName: String?,
    val staffRole: String?,
    @field:CdnImageUrl val staffPhoto: String?,
) {
    companion object {
        fun from(result: StaffResult): PerformanceDetailStaffResponse =
            PerformanceDetailStaffResponse(result.id, result.name, result.role, result.photo)
    }
}
